### PR TITLE
sl-71: Start stepper target audience

### PR DESCRIPTION
**Pull request by Alexandr-mas**
…ct, made a multiselect, changed the visual styles

### CODE REVIEW

File: frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { type ReactNode } from 'react';
+import { type OptionProps, components } from 'react-select';
+import {
+  type SiteCreateStepTarget,
+  siteCreateStepTargetValidationSchema,
+} from 'shared/build';
```
Comment (jaroslawtt) [inline frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx]: Should it actually be imported exact from shared/build?

File: frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import { type ReactNode } from 'react';
```
Comment (jaroslawtt) [inline frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx]: Can use React.ReactNode 
```suggestion

```

**Review by jaroslawtt — CHANGES_REQUESTED** [review]

**Comment by Alexandr-mas** [comment]
image
this is with validation
image
it's a multiselect

File: shared/src/packages/sites/libs/validation-schemas/site-create-step-target.validation-schema.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import joi from 'joi';
+
+import { SiteValidationMessage } from '../enums/enums.js';
+import { type SiteCreateStepTarget } from '../types/types.js';
+
+const siteCreateStepTarget = joi.object<SiteCreateStepTarget, true>({
+  targetAudience: joi
+    .array()
```
Comment (what1s1ove) [inline shared/src/packages/sites/libs/validation-schemas/site-create-step-target.validation-schema.ts]: should be an array of strings

File: frontend/src/pages/start/start.tsx [hunk 4]
```diff
@@ -11,10 +11,11 @@ import { type SiteCreateRequestDto } from '~/packages/sites/sites.js';
 import { actions as siteActions } from '~/slices/sites/sites.js';
 
 import { FinalForm, ProjectNameForm } from './libs/components/components.js';
+import { TargetAudience } from './libs/components/target-audience-form/target-audience-form.js';
```
Comment (what1s1ove) [inline frontend/src/pages/start/start.tsx]:
```suggestion
import { TargetAudienceForm } from './libs/components/target-audience-form/target-audience-form.js';
```
concat the imports

File: shared/src/packages/sites/libs/validation-schemas/site-create-step-target.validation-schema.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    .items(
+      joi.object({
+        label: joi.string().required(),
+        value: joi.string().required(),
+      }),
+    )
+    .min(1)
```
Comment (what1s1ove) [inline shared/src/packages/sites/libs/validation-schemas/site-create-step-target.validation-schema.ts]: add validation rule for all rules

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/libs/components/select/select.tsx [hunk 6]
```diff
@@ -18,18 +18,62 @@ type Properties<T extends FieldValues> = {
   options: SelectOption<string | number>[];
   placeholder?: string;
   errors: FieldErrors<T>;
+  isMulti?: boolean;
+  hideSelectedOptions: boolean;
+  customComponents?: object;
```
Comment (andanf-e) [inline frontend/src/libs/components/select/select.tsx]: First of all custom components in react-select may be typed,, and the second - you should provide customComponents as outer props - but just specify and pass needed ones if it's multiselect

File: shared/src/packages/sites/libs/types/site-create-request-dto.type.ts [hunk 7]
```diff
@@ -1,6 +1,7 @@
 type SiteCreateRequestDto = {
   name: string;
   industry: string;
+  targetAudience: { label: string; value: string }[];
```
Comment (andanf-e) [inline shared/src/packages/sites/libs/types/site-create-request-dto.type.ts]: But it should be array of strings

File: shared/src/packages/sites/libs/types/site-create-response-dto.type.ts [hunk 8]
```diff
@@ -1,6 +1,7 @@
 type SiteCreateResponseDto = {
   id: number;
   name: string;
+  targetAudience: { label: string; value: string }[];
```
Comment (andanf-e) [inline shared/src/packages/sites/libs/types/site-create-response-dto.type.ts]: Same about array of strings

File: shared/src/packages/sites/libs/enums/site-validation-message.enum.ts [hunk 9]
```diff
@@ -2,6 +2,8 @@ const SiteValidationMessage = {
   SITE_NAME_REQUIRE: 'Project name is required',
   SITE_NAME_WRONG: 'Project name is wrong',
   SITE_NAME_PATTERN: 'Incorrect project name',
+  SITE_TARGET_REQUIRE: 'Required field',
+  SITE_TARGET_WRONG: 'Invalid field',
```
Comment (andanf-e) [inline shared/src/packages/sites/libs/enums/site-validation-message.enum.ts]: What field is required and what is invalid -- you have specific keys and abstract messages

File: shared/src/packages/sites/libs/types/site-create-step-target.type.ts [hunk 10]
```diff
@@ -0,0 +1,4 @@
+type SiteCreateStepTarget = {
+  targetAudience: { label: string; value: string }[];
```
Comment (andanf-e) [inline shared/src/packages/sites/libs/types/site-create-step-target.type.ts]:
```suggestion
  targetAudience: string[];
```

File: shared/src/packages/sites/libs/types/site-get-all-item-response-dto.type.ts [hunk 11]
```diff
@@ -1,6 +1,7 @@
 type SiteGetAllItemResponseDto = {
   id: number;
   name: string;
+  targetAudience: { label: string; value: string }[];
```
Comment (andanf-e) [inline shared/src/packages/sites/libs/types/site-get-all-item-response-dto.type.ts]:
```suggestion
  targetAudience: string[];
```

File: frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx [hunk 12]
```diff
@@ -50,29 +50,9 @@ const TargetAudienceForm: React.FC<Properties> = ({
...
-                    }}
-                  />
-                  <div>{properties.label}</div>
-                </div>
-              </components.Option>
-            ),
-          }}
         />
         <Button
           label="Next"
```
Comment (Izdormu) [inline frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx]: Change label name to "Go next"

File: backend/src/packages/sites/site.service.ts [hunk 13]
```diff
@@ -42,7 +42,10 @@ class SiteService implements Omit<IService, 'find' | 'update' | 'delete'> {
     const sites = await this.siteRepository.findAll();
 
     return {
-      items: sites.map((site) => site.toObject()),
+      items: sites.map((site) => ({
+        ...site.toObject(),
+        targetAudience: [],
```
Comment (andanf-e) [inline backend/src/packages/sites/site.service.ts]: It's not a part of sites entity

File: frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx
Comment (andanf-e) [inline frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx:1]: Pay attention on warns about never used stuff

File: frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+        What is your Target audience?
+      </h2>
+      <div className={getValidClassNames(style['subtitle'])}>
+        Choose the age
+      </div>
+      <form className={style['form']} onSubmit={handleFormSubmit}>
+        <Select
+          control={control}
+          name="targetAudience"
+          isMulti
```
Comment (what1s1ove) [inline frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx]: should not a multiple select (according the documentation)

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Comment by Alexandr-mas** [comment]
image

**Review by jaroslawtt — APPROVED** [review]

File: backend/src/packages/sites/site.service.ts [hunk 15]
```diff
@@ -129,11 +129,13 @@ class SiteService implements Omit<IService, 'find' | 'update' | 'delete'> {
     const EXAMPLE_COMPANY_NAME = 'id Studio';
     const EXAMPLE_INDUSTRY = 'interior design';
     const EXAMPLE_TONE = 'official';
+    const EXAMPLE_TARGET = 'young-adult';
```
Comment (what1s1ove) [inline backend/src/packages/sites/site.service.ts]: use enum values here

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/packages/sites/site.service.ts [hunk 16]
```diff
@@ -1,3 +1,5 @@
+import { SiteTargetType } from 'shared/build/index.js';
```
Comment (Alyona-Pisotska) [inline backend/src/packages/sites/site.service.ts]: fix import

File: frontend/src/pages/start/libs/components/target-audience-form/style.module.scss [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+  font-weight: 400;
+  font-size: 14px;
+}
+
+.form {
+  display: flex;
+  flex-grow: 1;
+  flex-direction: column;
+  margin-top: 30px;
+  border-color: black;
```
Comment (Alyona-Pisotska) [inline frontend/src/pages/start/libs/components/target-audience-form/style.module.scss]: use a variable from variables.scss

File: frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx [hunk 18]
```diff
@@ -0,0 +1,60 @@
...
+  const handleFormSubmit = useCallback(
+    (event_: React.BaseSyntheticEvent): void => {
+      void handleSubmit(onSubmit)(event_);
+    },
+    [handleSubmit, onSubmit],
+  );
+
+  return (
+    <>
+      <h2 className={getValidClassNames(style['title'])}>
```
Comment (Alyona-Pisotska) [inline frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx]: You don't need `getValidClassNames ` there

File: frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+        What is your Target audience?
+      </h2>
+      <div className={getValidClassNames(style['subtitle'])}>
```
Comment (Alyona-Pisotska) [inline frontend/src/pages/start/libs/components/target-audience-form/target-audience-form.tsx]: same

**Review by Alyona-Pisotska — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**